python/mage/link_prediction/factory.py

```diff
@@ -0,0 +1,89 @@
+from typing import List, Tuple
+import torch
+from mage.link_prediction.models.GraphSAGE import GraphSAGE
+from mage.link_prediction.models.GAT import GAT
+from mage.link_prediction.predictors.DotPredictor import DotPredictor
+from mage.link_prediction.predictors.MLPPredictor import MLPPredictor
+from mage.link_prediction.constants import (
+    ADAM_OPT,
+    SGD_OPT,
+    GRAPH_ATTN,
+    GRAPH_SAGE,
+    DOT_PREDICTOR,
+    MLP_PREDICTOR,
+)
+import itertools
+
+
+def create_optimizer(
+    optimizer_type: str,
+    learning_rate: float,
+    model: torch.nn.Module,
+    predictor: torch.nn.Module,
+) -> torch.optim.Optimizer:
+    """Creates optimizer with given optimizer type and learning rate.
+
+    Args:
+        optimizer_type (str): Type of the optimizer.
+        learning_rate (float): Learning rate of the optimizer.
+        model (torch.nn.Module): A reference to the already created model. Needed to chain the parameters.
+        predictor: (torch.nn.Module): A reference to the already created predictor. Needed to chain the parameters.
+    Returns:
+        torch.nn.Optimizer: Optimizer used in the training.
+    """
+    if optimizer_type.upper() == ADAM_OPT:
+        return torch.optim.Adam(
+            itertools.chain(model.parameters(), predictor.parameters()),
+            lr=learning_rate,
+        )
+    elif optimizer_type.upper() == SGD_OPT:
+        return torch.optim.SGD(
+            itertools.chain(model.parameters(), predictor.parameters()),
+            lr=learning_rate,
+        )
+    else:
+        raise Exception(f"Optimizer {optimizer_type} not supported. ")
```

Comment:
Create a dictionary and check whether something exists:
```
optimizers: {ADAM_OPT: torch.optim.Adam, SGD_OPT: torch.optim.SGD}
opt_name = optimizer_type.upper()
if opt_name in optimizers:
        return optimizers[opt_name](
            itertools.chain(model.parameters(), predictor.parameters()),
            lr=learning_rate,
        )
else:
    ...
```

Reply:
Just something to think of

Reply:
jump table pattern right? Very cool!